examples/angular/src/features/hacker-news-story/hacker-news-story.provider.ts

```diff
@@ -23,8 +23,8 @@ import { HackerNewsStoryIdsNetworkDataSource } from "./data/data-sources/hacker-
 const Cached = createCacheDecorator();
 
 export abstract class HackerNewsStoryProvider {
-  abstract getHackerNewsLatestAskStories(): GetHackerNewsLatestAskStoriesInteractor;
-  abstract getHackerNewsStory(): GetHackerNewsStoryInteractor;
+  abstract getGetHackerNewsLatestAskStories(): GetHackerNewsLatestAskStoriesInteractor;
+  abstract getGetHackerNewsStory(): GetHackerNewsStoryInteractor;
```

Comment:
This is "correct", but the naming makes it hard to read/understand what's going on here. I propose that we change the naming pattern when using Factory. Here are some options:

- createGetHackerNewsStory
- provideGetHackerNewsStory
- fetchGetHackerNewsStory

Thoughts? Preferences ? @orioljp @doup

Reply:
@lucianosantana  @doup @orioljp I would use `Factory` -> `GetHackerNewsStoryFactory`, as is a well-known naming in the industry for this kind of use.

Reply:
Being a method, it reads better if named as an "action". @Alex-DA proposal would make sense to me if we had a class named `GetHackerNewsStoryFactory` and then we had a method to call, such as: create, execute, etc.

It seems "create" is quite common for Factories.

- https://refactoring.guru/design-patterns/factory-method
- https://refactoring.guru/design-patterns/abstract-factory

Reply:
I wouldn't complicate it because you already know what's happening. After all, it's a function within the `Provider` which is the DI class, so you are aware that it's creating a new instance of the class. 

If you want a singleton, you can not create an `abstract` function and just define a unique variable. 

A factory or abstract factory is when you want to create and instantiate different implementations from the class/interface dynamically, which we are not doing it here. We are statically creating the instance in the `Provider`, maybe in some cases, we do it dynamically but not from the caller (component, presenter, interaction, etc.); we do it given a configuration in the `Provider` constructor. 

The key here is that the function or abstract function in within the `Provider` which you already know that it's the DI class for instantiating objects.

Reply:
I agree. I mean, I wasn't suggesting to use an abstract factory or to rename the class, but just to borrow the **naming pattern**. The name `getGetHackerNewsStory` is weird to read, sounds like cacophony, and might distract from understanding the meaning. 

In the links I've shared, the method names, both for Factory Method and Abstract Factory patterns, are using the "create" prefix. Hence, my first suggestion was using something like `createGetHackerNewsStory `. 

The second suggestion was `provideGetHackerNewsStory`, since the class name is already a "Provider" and the expected action is to "provide" something. 

The third, "fetch", is just a synonym of "get". It's not the best one, in my opinion, but is better than `getGetSomething`.

Reply:
I think we are entering too into detail for something that it's still needed to deep dive by the web team. 

Happy any of the options, we do not have a strict rule in the Mobile Team because it does not add a lot of value *right now*

Reply:
In this case I'd use `provide` as prefix, the class is already a Provider and I believe there's less probability of cacophony naming.

Reply:
My bid for `provide` or `factory`.

Reply:
We cannot use `action` as @lucianosantana suggests, this is no an action, undestanding an action something that the user triggers.

I don't think it's wrong calling getXXX since it's what it is, but since it's the only method that knows how to instanciate an interactor I think the same as @Alex-DA .

Since `factory` is not a prefix, but a sufix, and getXXXXXFactory would mean that we're getting a factory, the best option is provideGetXXXXX,

Reading your comment about finding names that can collide @doup ... I'm sure you can find an alternative for ProvideConfigurationInteractor that does not use Provide as a prefix ;).